File: A/another_string_minimization_problem.py
for _ in range(int(input())):
    n, m = map(int, input().split())
    a = list(map(int, input().split()))
    s = ['B'] * m
    for x in a:
        y = m - x
        x = x - 1
        l, r = min(x, y), max(x, y)
        s[l if s[l] == 'B' else r] = 'A'
    print(''.join(s))

File: A/another_string_minimization_problem.cpp
#include <bits/stdc++.h>

using namespace std;

void solve() {
    int n, m, a, b;
    cin >> n >> m;
    string s(m, 'B');
    for (int i = 0; i < n; ++i) {
        cin >> a;
        b = m - a--;
        int l = min(a, b);
        int r = max(a, b);
        s[l] == 'B' ? s[l] = 'A' : s[r] = 'A';
    }
    cout << s << '\n';
}

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    int t;
    cin >> t;
    while (t--) {
        solve();
    }
    return 0;
}
